Automattic/wp-calypso
# Post Normalizer: Refactor rules into individual files

And make the sync rules truly sync, wrapping them in the main post normalizer.

This really just moves all of the post normalizer rules out of the one massive file into little files. This helps scoping quite a bit and makes it clearer what each rule does and what constants it depends on.

I also took the opportunity to make all of the sync rules look and feel like sync functions. No longer do they take a callback, which implies async behavior, and invoke is synchronously. There is a new wrapper that converts the sync variants to look like the old async-ish version so the refactoring doesn't impact users of the post normalizer. 

To test, 
1. Run the test suite. Yay tests!
2. Try using the Reader. Posts should categorize based on content, like they do now. Posts without a featued image spec'd should get one when a good candidate is found
3. Try using My Posts. Same basic set of criteria.

Stuff for another PR
- [ ] Rework lib/feed-post-store/* to use the rules directly and avoid using postNormalizer and async
- [ ] Rework lib/posts to use the rules directly
- [ ] Find anywhere else that's using post norm and rework it
- [ ] Kill post norm and the async dependency
